File: Container With Most Water/mostWaterContainer.cpp
#include<bits/stdc++.h>
using namespace std;

int mostWaterContainer(int arr[], int n)
{
    // Function to calculate maximum area of water that can be contained
    int left = 0;
    int right = n - 1;
    int area = 0;

    while (left < right)
    {
        area = max(area, min(arr[left], arr[right]) * (right - left));

        if (arr[left] < arr[right])
            left += 1;

        else
            right -= 1;
    }
    return area;
}
int main() {
    int n = 9;  // Size of the array
    int arr[] = {1, 8, 6, 2, 5, 4, 8, 3, 7};

    cout << mostWaterContainer(arr, n) << endl;
    return 0;
}

File: 23_Container With Most Water/Ideal Solutions/mostWaterContainer.py

# Calculating the conatainer area with most water
def mostWaterContainer(container):

    # Setting left pointer, right pointer and area
    leftMostSide = 0
    rightMostSide = len(container) - 1
    maxArea = 0

    while leftMostSide < rightMostSide:

        # Keep Track of max area
        maxArea = max(maxArea, min(container[leftMostSide], container[rightMostSide]) * (rightMostSide - leftMostSide))

        # Either side which will be smaller , index will be updated
        if container[leftMostSide] < container[rightMostSide]:
            leftMostSide += 1
        else:
            rightMostSide = rightMostSide - 1

    return maxArea


if __name__ == "__main__":
    array = []

    # Inputs
    container = [1, 8, 6, 2, 5, 4, 8, 3, 7]

    print(mostWaterContainer(container))
